src/context/MovieContext.jsx

```diff
@@ -0,0 +1,135 @@
+/* eslint-disable default-param-last */
+import React, { useReducer, createContext, useMemo, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import cloneDeep from 'lodash/cloneDeep';
+
+import { movieData } from '../mock_data';
+
+export const MovieContext = createContext({});
+
+export const Actions = {
+  TO_EDIT: 'TO_EDIT',
+  TO_ADD: 'TO_ADD',
+  TO_DELETE: 'TO_DELETE',
+  OPEN_DESCRIPTION: 'OPEN_DESCRIPTION',
+  CLOSE_DESCRIPTION: 'CLOSE_DESCRIPTION'
+};
+
+const initialState = {
+  triggerDescription: false,
+  id: '',
+  moviesData: cloneDeep(movieData),
+  movieContent: {
+    title: '',
+    movieType: '',
+    id: '',
+    image: '',
+    year: '',
+    rating: '',
+    runtime: '',
+    description: ''
+  }
+};
+
+const ManageModalReducer = (state, action) => {
+  const { type, payload } = action;
+  const arrayToModify = cloneDeep(state.moviesData);
+  switch (type) {
+    case Actions.TO_EDIT: {
+      return {
+        ...state,
+        ...{
+          moviesData: arrayToModify.map((movie) => {
+            if (movie.id === payload.id) {
+              return payload;
+            }
+            return movie;
+          })
+        }
+      };
+    }
+    case Actions.TO_ADD: {
```

Comment:
Add movie logic doesn't work, I think this should help:

![image](https://user-images.githubusercontent.com/52933810/161712419-134e841d-304c-47c7-a8d4-3997a9b9ebef.png)